app/src/common/shared/com/igalia/wolvic/browser/engine/Session.java

```diff
@@ -72,7 +74,10 @@ public class Session implements WContentBlocking.Delegate, WSession.NavigationDe
     private static UriOverride sUserAgentOverride;
     private static UriOverride sDesktopModeOverrides;
     private static final long KEEP_ALIVE_DURATION_MS = 1000; // 1 second.
-    private static final String DEFAULT_USER_AGENT = "Mozilla/5.0 (Android 10; Mobile VR; rv:105.0) Gecko/105.0 Wolvic/" + BuildConfig.VERSION_NAME;
+
+    // The difference between "Mobile" and "VR" matches GeckoViewSettings.jsm
+    private static final String WOLVIC_USER_AGENT_MOBILE = GeckoSession.getDefaultUserAgent() + " Wolvic/" + BuildConfig.VERSION_NAME;
```

Comment:
We cannot do this, this file is engine agnostic, and that's why you could not use GeckoSession here. You might want to add a new method to WSession and then implement it in the existing SessionImpl.java instances (you could leave the chromium one unimplemented).

Reply:
Thank you. I have moved this logic to SessionImpl.java.